### PR TITLE
New BLE library integration

### PR DESCRIPTION
**Pull request by bmaczak**

### CODE REVIEW

File: app/src/main/java/org/revolutionrobotics/robotcontroller/bluetooth/connect/ConnectDialog.kt [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    fun connect(device: Device) {
+        if (!connecting) {
+            connecting = true
+            deviceDiscoverer.stopDiscovering()
+            adapter.showConnecting(device)
+            getExampleActivity().deviceConnector.connect(device, onConnected = {
+                Toast.makeText(activity, "Connected", Toast.LENGTH_SHORT).show()
+                dismiss()
+            }, onError = {
+                Toast.makeText(activity, "Error", Toast.LENGTH_SHORT).show()
```
Comment (bugadani) [inline app/src/main/java/org/revolutionrobotics/robotcontroller/bluetooth/connect/ConnectDialog.kt]: Can we log this somehow?

File: bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+        return if (startIndex < message.size) {
+            val endIndex = (startIndex + dataLength).coerceAtMost(message.size)
+            val chunk = ByteArray(endIndex - startIndex + 1).apply {
+                this[0] = firstByte
+                message.copyInto(this, 1, startIndex, endIndex)
+            }
+            Log.d(RoboticsConfigurationService.TAG, "Write message sent: ${chunk.toStringCustom()}")
+            chunk
+        } else {
+            null
```
Comment (bugadani) [inline bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt]: Is this a good idea? This case smells like an exception instead

File: app/src/main/java/org/revolutionrobotics/robotcontroller/bluetooth/connect/ConnectDialog.kt (hunk 1 above)
Comment (bmaczak) [reply app/src/main/java/org/revolutionrobotics/robotcontroller/bluetooth/connect/ConnectDialog.kt]: This is only part of the test app, do you think logging woulb

File: bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt (hunk 2 above)
Comment (bmaczak) [reply bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt]: This is the way the DataSplitter interface works. From the docs:
@return The packet to be sent, or null, if the whole message was already split.

File: app/src/main/res/layout/acrtivity_example.xml [hunk 3]
```diff
@@ -1,228 +1,242 @@
 <?xml version="1.0" encoding="utf-8"?>
```
Comment (vhermecz) [inline app/src/main/res/layout/acrtivity_example.xml]: Fun of autocomplete, note the filename :)

File: bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt [hunk 4]
```diff
@@ -0,0 +1,29 @@
+package org.revolutionrobotics.bluetooth.android.communication
+
+import android.util.Log
+import no.nordicsemi.android.ble.data.DataSplitter
+import org.revolutionrobotics.bluetooth.android.service.RoboticsConfigurationService
+
+class LongMessageSplitter(
+    private val firstByte: Byte,
```
Comment (vhermecz) [inline bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt]: Yes, we see that this is the first byte, but what is the semantics?
As I recall, this is come kind of control character. Could you add a note about this one? Thx

File: bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    private val chunkLength: Int // for some reason setting the MTU doesn't work on some devices
+) : DataSplitter {
+
+    override fun chunk(message: ByteArray, index: Int, maxLength: Int): ByteArray? {
```
Comment (vhermecz) [inline bluetooth-android/src/main/java/org/revolutionrobotics/bluetooth/android/communication/LongMessageSplitter.kt]: If I got it right, maxLength is sometimes corrupt? That is why you pass the chunkLength in the ctor?
Addin a note in the body about this I would love :) thx

**Pull request merged**